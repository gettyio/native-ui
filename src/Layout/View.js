import styled, { css } from 'styled-components/native'

const margin = css`
  ${props => props.m && css`margin: ${props.theme.spacing[props.m]}px;`}
  ${props => props.mh && css`margin-horizontal: ${props.theme.spacing[props.mh]}px;`}
  ${props => props.mv && css` margin-vertical: ${props.theme.spacing[props.mv]}px;`}
  ${props => props.mt && css` margin-top: ${props.theme.spacing[props.mt]}px;`}
  ${props => props.mb && css` margin-bottom: ${props.theme.spacing[props.mb]}px;`}
  ${props => props.ml && css` margin-left: ${props.theme.spacing[props.ml]}px;`}
  ${props => props.mr && css` margin-right: ${props.theme.spacing[props.mr]}px;`};
`
const padding = css`
  ${props => props.p && css`padding: ${props.theme.spacing[props.p]}px;`}
  ${props => props.ph && css`padding-horizontal: ${props.theme.spacing[props.ph]}px;`}
  ${props => props.pv && css` padding-vertical: ${props.theme.spacing[props.pv]}px;`}
  ${props => props.pt && css` padding-top: ${props.theme.spacing[props.pt]}px;`}
  ${props => props.pb && css` padding-bottom: ${props.theme.spacing[props.pb]}px;`}
  ${props => props.pl && css` padding-left: ${props.theme.spacing[props.pl]}px;`}
  ${props => props.pr && css` padding-right: ${props.theme.spacing[props.pr]}px;`};
`

const View = styled.View`
  ${props => props.flex && css`flex: ${props.flex};`}
  ${props => props.align && css`align-items: ${props.align};`}
  ${props => props.justify && css`justify-content: ${props.justify};`}
  ${props => props.background && css`background: ${props.background};`}
  ${props => props.height && css`height: ${props.height}px;`}
  ${props => props.width && css`width: ${props.width}px;`}
  ${margin}
  ${padding}
`

View.defaultProps = {
  theme: {
    spacing: [2, 4, 8, 16, 32, 64]
  }
}

export default View
