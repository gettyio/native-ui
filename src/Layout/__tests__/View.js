import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import 'jest-styled-components'

import View from '../View'

describe('View component', () => {
  it('should render without issue', () => {
    const component = shallow(<View />)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
