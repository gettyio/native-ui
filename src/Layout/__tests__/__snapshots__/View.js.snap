// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component has a custom align-items styled applied 1`] = `
<Component
  align="center"
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom background styled applied 1`] = `
<Component
  background="blue"
  style={
    Array [
      Object {
        "backgroundColor": "blue",
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom flex style applied 1`] = `
<Component
  flex="1"
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom height styled applied 1`] = `
<Component
  height={100}
  style={
    Array [
      Object {
        "height": 100,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom justify-content styled applied 1`] = `
<Component
  justify="center"
  style={
    Array [
      Object {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom margin styled applied 1`] = `
<Component
  m={10}
  style={
    Array [
      Object {
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom margin-bottom styled applied 1`] = `
<Component
  mb={10}
  style={
    Array [
      Object {
        "marginBottom": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom margin-horizontal styled applied 1`] = `
<Component
  mh={10}
  style={
    Array [
      Object {
        "marginHorizontal": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom margin-left styled applied 1`] = `
<Component
  ml={10}
  style={
    Array [
      Object {
        "marginLeft": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom margin-right styled applied 1`] = `
<Component
  mr={10}
  style={
    Array [
      Object {
        "marginRight": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom margin-top styled applied 1`] = `
<Component
  mt={10}
  style={
    Array [
      Object {
        "marginTop": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom margin-vertical styled applied 1`] = `
<Component
  mv={10}
  style={
    Array [
      Object {
        "marginVertical": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom padding styled applied 1`] = `
<Component
  p={10}
  style={
    Array [
      Object {
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom padding-bottom styled applied 1`] = `
<Component
  pb={10}
  style={
    Array [
      Object {
        "paddingBottom": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom padding-horizontal styled applied 1`] = `
<Component
  ph={10}
  style={
    Array [
      Object {
        "paddingHorizontal": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom padding-left styled applied 1`] = `
<Component
  pl={10}
  style={
    Array [
      Object {
        "paddingLeft": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom padding-right styled applied 1`] = `
<Component
  pr={10}
  style={
    Array [
      Object {
        "paddingRight": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom padding-top styled applied 1`] = `
<Component
  pt={10}
  style={
    Array [
      Object {
        "paddingTop": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom padding-vertical styled applied 1`] = `
<Component
  pv={10}
  style={
    Array [
      Object {
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`View component has a custom width styled applied 1`] = `
<Component
  style={
    Array [
      Object {
        "width": 100,
      },
      undefined,
    ]
  }
  width={100}
/>
`;

exports[`View component should render without issue 1`] = `
<Component
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;
